tests/schema/tests.py

```diff
@@ -2454,6 +2454,85 @@ def test_func_unique_constraint_nondeterministic(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        constraint = UniqueConstraint(
+            fields=['name'],
+            name='author_name_tablespace_unq',
+            db_tablespace='pg_default',
+        )
+        # Add constraint.
+        with connection.schema_editor() as editor:
+            editor.add_constraint(Author, constraint)
+            sql = constraint.create_sql(Author, editor)
+        table = Author._meta.db_table
+        constraints = self.get_constraints(table)
+        self.assertIn(constraint.name, constraints)
+        self.assertIs(constraints[constraint.name]['unique'], True)
+        # SQL contains column and a tablespace
+        self.assertIs(sql.references_column(table, 'name'), True)
+        self.assertIn('TABLESPACE %s' % editor.quote_name(constraint.db_tablespace), str(sql))
+        # Remove constraint.
+        with connection.schema_editor() as editor:
+            editor.remove_constraint(Author, constraint)
+        self.assertNotIn(constraint.name, self.get_constraints(table))
+
+    @skipUnlessDBFeature(
+        'supports_expression_indexes',
+        'supports_tablespaces',
+        'supports_partial_indexes',
+        'supports_covering_indexes',
+        'supports_order_by_nulls_modifier',
+    )
+    def test_unique_constraint_db_tablespace_with_conditional_covering(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        constraint = UniqueConstraint(
+            F("name").desc(nulls_last=True),
+            name='author_name_desc_tbls_cond_inc_unq',
+            condition=Q(weight__isnull=False),
+            include=['weight'],
+            db_tablespace='pg_default',
+        )
+        # Add constraint.
+        with connection.schema_editor() as editor:
+            editor.add_constraint(Author, constraint)
+            sql = constraint.create_sql(Author, editor)
+        table = Author._meta.db_table
+        constraints = self.get_constraints(table)
+        self.assertIn(constraint.name, constraints)
+        self.assertIs(constraints[constraint.name]['unique'], True)
+        self.assertEqual(
+            constraints[constraint.name]['columns'],
+            ['name', 'weight'],
+        )
+        if connection.features.supports_index_column_ordering:
+            self.assertIndexOrder(table, constraint.name, ['DESC'])
+        # SQL contains column, a condition and a tablespace
+        self.assertIs(sql.references_column(table, 'name'), True)
+        self.assertIn(
+            'WHERE %s IS NOT NULL' % editor.quote_name('weight'),
+            str(sql),
+        )
+        self.assertIn('TABLESPACE %s' % editor.quote_name(constraint.db_tablespace), str(sql))
+        # Remove constraint.
+        with connection.schema_editor() as editor:
+            editor.remove_constraint(Author, constraint)
+        self.assertNotIn(constraint.name, self.get_constraints(table))
```

Comment:
These tests will not work on Oracle, we would need to create a custom tablespace for them. I think it's enough to check that a properly quoted tablespace name is in the generated SQL like we do for indexes, see 
https://github.com/django/django/blob/e513fb0e77baf2ebcbf2cbe366bdf0228d01119f/tests/model_indexes/tests.py#L286-L320

Reply:
Thanks! I guess we can still use the PostgreSQL test `test_opclass_tablespace` to still test the actual creation with the `pg_default` tablespace.